src/main/java/com/alibou/security/auth/AuthenticationService.java

```diff
@@ -32,14 +32,18 @@ public class AuthenticationService {
   private final AuthenticationManager authenticationManager;
 
   public AuthenticationResponse register(RegisterRequest request) {
-    var user = User.builder()
-        .firstname(request.getFirstname())
-        .lastname(request.getLastname())
-        .email(request.getEmail())
-        .password(passwordEncoder.encode(request.getPassword()))
-        .role(request.getRole())
-        .build();
-    var savedUser = repository.save(user);
+    // Check if the given user exists in database
+    boolean isPresent = repository.findByEmail(request.getEmail()).isPresent();
+    var user = isPresent
+            ? repository.findByEmail(request.getEmail()).get()
```

Comment:
You are fetching the same data twice from DB?

Reply:
Thanks for your feedback :) Bug fixed